File: Household Budget Calculator/[VB]-Household Budget Calculator/VB/Household budget calculator/expanding property classes/OtherBills.vb
Imports System.ComponentModel

<TypeConverter(GetType(ExpandableObjectConverter))> <System.Serializable()> _
Public Class OtherBills

    Public Event propertiesChanged(ByVal e As propertiesChangedEventArgs)

    Private _cc As Decimal
    <DisplayName("Credit Cards"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property cc() As Decimal
        Get
            Return _cc
        End Get
        Set(ByVal value As Decimal)
            _cc = value
            totalOtherBills = cc + loans + maintenance + opticalAndDental + miscellaneous
        End Set
    End Property

    Private _loans As Decimal
    <DisplayName("Loan Repayments"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property loans() As Decimal
        Get
            Return _loans
        End Get
        Set(ByVal value As Decimal)
            _loans = value
            totalOtherBills = cc + loans + maintenance + opticalAndDental + miscellaneous
        End Set
    End Property

    Private _maintenance As Decimal
    <DisplayName("Maintenance"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property maintenance() As Decimal
        Get
            Return _maintenance
        End Get
        Set(ByVal value As Decimal)
            _maintenance = value
            totalOtherBills = cc + loans + maintenance + opticalAndDental + miscellaneous
        End Set
    End Property

    Private _opticalAndDental As Decimal
    <DisplayName("Optical/Dental Costs"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property opticalAndDental() As Decimal
        Get
            Return _opticalAndDental
        End Get
        Set(ByVal value As Decimal)
            _opticalAndDental = value
            totalOtherBills = cc + loans + maintenance + opticalAndDental + miscellaneous
        End Set
    End Property

    Private _miscellaneous As Decimal
    <DisplayName("Miscellaneous"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property miscellaneous() As Decimal
        Get
            Return _miscellaneous
        End Get
        Set(ByVal value As Decimal)
            _miscellaneous = value
            totalOtherBills = cc + loans + maintenance + opticalAndDental + miscellaneous
        End Set
    End Property

    Private _totalOtherBills As Decimal
    <ReadOnlyAttribute(True), _
    Browsable(False), _
    DisplayName("Total Other Bills Outgoings")> _
    Public Property totalOtherBills() As Decimal
        Get
            Return _totalOtherBills
        End Get
        Set(ByVal value As Decimal)
            _totalOtherBills = value
            RaiseEvent propertiesChanged(New propertiesChangedEventArgs With {.propName = "Other Bills", .newValue = value})
        End Set
    End Property

    Public Overrides Function ToString() As String
        Return totalOtherBills.ToString("c2")
    End Function

End Class

File: Household Budget Calculator/[C#]-Household Budget Calculator/C#/Household budget calculator CS/expanding property classes/OtherBills.cs
//using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.ComponentModel;

namespace Household_budget_calculator_CS.expanding_property_classes
{
    [TypeConverter(typeof(ExpandableObjectConverter))]
    [System.Serializable()]
    public class OtherBills
    {
        public event propertiesChangedEventHandler propertiesChanged;
        public delegate void propertiesChangedEventHandler(propertiesChangedEventArgs e);

        private decimal _cc;
        [DisplayName("Credit Cards"), RefreshProperties(RefreshProperties.All)]
        public decimal cc
        {
            get { return _cc; }
            set
            {
                _cc = value;
                totalOtherBills = cc + loans + maintenance + opticalAndDental + miscellaneous;
            }
        }

        private decimal _loans;
        [DisplayName("Loan Repayments"), RefreshProperties(RefreshProperties.All)]
        public decimal loans
        {
            get { return _loans; }
            set
            {
                _loans = value;
                totalOtherBills = cc + loans + maintenance + opticalAndDental + miscellaneous;
            }
        }

        private decimal _maintenance;
        [DisplayName("Maintenance"), RefreshProperties(RefreshProperties.All)]
        public decimal maintenance
        {
            get { return _maintenance; }
            set
            {
                _maintenance = value;
                totalOtherBills = cc + loans + maintenance + opticalAndDental + miscellaneous;
            }
        }

        private decimal _opticalAndDental;
        [DisplayName("Optical/Dental Costs"), RefreshProperties(RefreshProperties.All)]
        public decimal opticalAndDental
        {
            get { return _opticalAndDental; }
            set
            {
                _opticalAndDental = value;
                totalOtherBills = cc + loans + maintenance + opticalAndDental + miscellaneous;
            }
        }

        private decimal _miscellaneous;
        [DisplayName("Miscellaneous"), RefreshProperties(RefreshProperties.All)]
        public decimal miscellaneous
        {
            get { return _miscellaneous; }
            set
            {
                _miscellaneous = value;
                totalOtherBills = cc + loans + maintenance + opticalAndDental + miscellaneous;
            }
        }

        private decimal _totalOtherBills;
        [ReadOnlyAttribute(true), Browsable(false), DisplayName("Total Other Bills Outgoings")]
        public decimal totalOtherBills
        {
            get { return _totalOtherBills; }
            set
            {
                _totalOtherBills = value;
                if (propertiesChanged != null)
                {
                    propertiesChanged(new propertiesChangedEventArgs
                    {
                        propName = "Other Bills",
                        newValue = value
                    });
                }
            }
        }

        public override string ToString()
        {
            return totalOtherBills.ToString("c2");
        }

    }
}
